vispy/visuals/filters/mesh.py

```diff
@@ -134,17 +134,17 @@ def _attach(self, visual):
     vec3 normal = normalize(normal_scene.xyz - origin_scene.xyz);
     v_normal_vec = normal;
 
+    // Calculate eye vector in a vertex-depending manner. I.e. consider the eye to
+    // occupy a rectangular plane.
     vec4 pos_front = $scene2doc(pos_scene);
-    pos_front.z += 1e-6;
-    pos_front = $doc2scene(pos_front);
     pos_front /= pos_front.w;
-
-    vec4 pos_back = $scene2doc(pos_scene);
-    pos_back.z -= 1e-6;
+    vec4 pos_back = pos_front;
+    pos_front.z -= 1e-5;
+    pos_back.z += 1e-5;
+    pos_front = $doc2scene(pos_front);
```

Comment:
So, the main change is here, right? The eye vector is created in normalised coordinates and is thus now independent of the size of the scene/frustum/models. Right?

Reply:
The main change is actually turning those `1e6` into `1e5`. There is a bit of re-ordering to do the math with less transforms .

Reply:
I see, changing to `1e5` only suffices, but I still have no evidence that this is not specific to this example. I wonder if a perspective correction would not be a more general solution. Examples:

No perspective correction (`1e6` too small relative to the size of the viewing frustum):
```c
    // Fails: 'glittering' artefacts when not close enough to the tubes in `examples/basics/visuals/tube.py`.
    vec4 pos_eye = $scene2doc(pos_scene);

    vec4 pos_front = pos_eye;
    pos_front.z += 1e-6;
    pos_front = $doc2scene(pos_front);
    pos_front /= pos_front.w;

    vec4 pos_back = pos_eye;
    pos_back.z -= 1e-6;
    pos_back = $doc2scene(pos_back);
    pos_back /= pos_back.w;
```

No perspective correction (`1e5` is big enough):
```c
    // Works.
    vec4 pos_eye = $scene2doc(pos_scene);

    vec4 pos_front = pos_eye;
    pos_front.z += 1e-5;
    pos_front = $doc2scene(pos_front);
    pos_front /= pos_front.w;

    vec4 pos_back = pos_eye;
    pos_back.z -= 1e-5;
    pos_back = $doc2scene(pos_back);
    pos_back /= pos_back.w;
```

With perspective correction, variant 1:
```c
    // Works.
    vec4 pos_eye = $scene2doc(pos_scene);

    pos_eye /= pos_eye.w;

    vec4 pos_front = pos_eye;
    pos_front.z += 1e-6;
    pos_front = $doc2scene(pos_front);
    pos_front /= pos_front.w;

    vec4 pos_back = pos_eye;
    pos_back.z -= 1e-6;
    pos_back = $doc2scene(pos_back);
    pos_back /= pos_back.w;
```

With perspective correction, variant 2 (the math is equivalent to variant 1):
```c
    // Works.
    vec4 pos_eye = $scene2doc(pos_scene);

    vec4 pos_front = pos_eye;
    pos_front.z += 1e-6 * pos_front.w;
    pos_front = $doc2scene(pos_front);
    pos_front /= pos_front.w;

    vec4 pos_back = pos_eye;
    pos_back.z -= 1e-6 * pos_back.w;
    pos_back = $doc2scene(pos_back);
    pos_back /= pos_back.w;
```

I'll try to check that in actual example tests.

Reply:
To summarise, I think this change makes things more correct and would be good to merge.